### ISSUE TITLE
Gollum does not follow page history through renames 

### ISSUE THREAD

**Issue by dekimsey** [gollum/gollum#565]
Currently if a page is renamed the history display of the new page stops short at the rename event. I'd very much like to see it follow and show the appropriate history events for the old page names.

**Comment by dometto** [comment]
This actually currently works on all adapters, even `grit`. However, unfortunately, it's excruciatingly slow on `rjgit`. All that's required to get the functionality to work is this minor change to `app.rb` though:

```diff
diff --git a/lib/gollum/app.rb b/lib/gollum/app.rb
index 2f0611f..94316a9 100644
--- a/lib/gollum/app.rb
+++ b/lib/gollum/app.rb
@@ -408,7 +408,7 @@ module Precious
       @page     = wiki_page(params[:splat].first).page
       @page_num = [params[:page].to_i, 1].max
       unless @page.nil?
-        @versions = @page.versions :page => @page_num
+        @versions = @page.versions(:page => @page_num, :follow => true)
```

We could enable this by default and allow an option to disable `:follow` through `config.rb`, or vice versa?

**Comment by bartkamphorst** [comment]
> it's excruciatingly slow on rjgit

Curious. Is it the rjgit adapter or RJGit or JGit that causes the slowdown, do you think?

> We could enable this by default and allow an option to disable :follow through config.rb, or vice versa?

Good idea.

**Comment by dometto** [comment]
> Is it the rjgit adapter or RJGit or JGit that causes the slowdown, do you think?

Either RJGit or JGit, as using RJGit, this:
```ruby
jruby-9.1.8.0 :006 > repo.git.log("renamed.md", "HEAD", :follow => true)
```
...is equally slow.

**Comment by dometto** [comment]
Implemented in #1225, so upcoming in `v5.0`.
